File: 백준/Silver/21921. 블로그/블로그.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");
let [n , k] = input[0].split(" ").map(Number);
let arr = input[1].split(" ").map(Number);

let sum = 0;
for(let i = 0; i < n; i++){
  if(i < k) sum += arr[i];
}

let max = sum;
let left = 0;
let right = k - 1;
let count = 1;
while(true){
  left++;
  right++;
  if(right > n - 1) break;
  sum = sum + arr[right] - arr[left - 1];
  if(sum === max) count++;
  else if(sum > max){
    max = sum;
    count = 1;
  }
}
if(max === 0) console.log("SAD");
else console.log(max + "\n" + count);

File: 백준/Silver/21921. 블로그/블로그.py
N, X = map(int, input().split())
arr = list(map(int, input().split()))


cur_sum = sum(arr[:X])
max_sum = cur_sum
count = 1

for i in range(1, N - X + 1):
    cur_sum = cur_sum - arr[i - 1] + arr[i + X - 1]

    if cur_sum > max_sum:
        max_sum = cur_sum
        count = 1
    elif cur_sum == max_sum:
        count +=1


if max_sum == 0:
    print('SAD')
else:
    print(max_sum)
    print(count)